### PR TITLE
Fix(TDI-43770):upgrade httpcore,httpclient,commons-logging,codec

### PR DESCRIPTION
**Pull request by vyu-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-43770

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tBonitaDeploy/tBonitaDeploy_java.xml [hunk 1]
```diff
@@ -238,20 +239,21 @@
 			<IMPORT NAME="xpp3-1.1.6.jar" MODULE="xpp3-1.1.6.jar" MVN="mvn:org.talend.libraries/xpp3-1.1.6/6.0.0" UrlPath="platform:/plugin/org.talend.libraries.xml/lib/xpp3-1.1.6.jar" REQUIRED_IF="(DB_VERSION=='BONITA_652') AND (CLIENT_MODE=='JAVA_CLIENT')" />
 			
 			<IMPORT NAME="httpmime-4.2.5.jar" MODULE="httpmime-4.2.5.jar" MVN="mvn:org.talend.libraries/httpmime-4.2.5/6.0.0" UrlPath="platform:/plugin/org.talend.libraries.apache.http/lib/httpmime-4.2.5.jar" REQUIRED_IF="(DB_VERSION=='BONITA_652') AND (CLIENT_MODE=='JAVA_CLIENT')" />
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBonitaDeploy/tBonitaDeploy_java.xml]: this one still 4.2.5 ?

File: main/plugins/org.talend.designer.components.localprovider/components/tBonitaDeploy/tBonitaDeploy_java.xml [hunk 2]
```diff
@@ -238,20 +239,21 @@
...
+			<IMPORT NAME="commons-codec-1.11.jar" MODULE="commons-codec-1.11.jar" MVN="mvn:commons-codec/commons-codec/1.11" REQUIRED_IF="(DB_VERSION=='BONITA_652') AND (CLIENT_MODE=='JAVA_CLIENT')" />
 			
 			<!-- Bonita 7.2.4 HTTP API -->
 			<IMPORT NAME="talend-bonita-client-1.0.0.jar" MODULE="talend-bonita-client-1.0.0.jar" MVN="mvn:org.talend.libraries/talend-bonita-client-1.0.0/6.0.0" REQUIRED_IF="(DB_VERSION=='BONITA_724') AND (CLIENT_MODE=='HTTP_CLIENT')" />
 			<IMPORT NAME="bonita-client-7.2.4.jar" MODULE="bonita-client-7.2.4.jar" MVN="mvn:org.talend.libraries/bonita-client-7.2.4/6.3.0" REQUIRED_IF="(DB_VERSION=='BONITA_724') AND (CLIENT_MODE=='HTTP_CLIENT')" />
 			<IMPORT NAME="bonita-common-7.2.4.jar" MODULE="bonita-common-7.2.4.jar" MVN="mvn:org.talend.libraries/bonita-common-7.2.4/6.3.0" REQUIRED_IF="(DB_VERSION=='BONITA_724') AND (CLIENT_MODE=='HTTP_CLIENT')" />
 			
 			<IMPORT NAME="httpmime-4.2.5.jar" MODULE="httpmime-4.2.5.jar" MVN="mvn:org.talend.libraries/httpmime-4.2.5/6.0.0" UrlPath="platform:/plugin/org.talend.libraries.apache.http/lib/httpmime-4.2.5.jar" REQUIRED_IF="(DB_VERSION=='BONITA_724') AND (CLIENT_MODE=='HTTP_CLIENT')" />
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBonitaDeploy/tBonitaDeploy_java.xml]: this too.

File: main/plugins/org.talend.designer.components.localprovider/components/tBoxConnection/tBoxConnection_java.xml [hunk 3]
```diff
@@ -81,20 +81,20 @@
...
             <IMPORT NAME="apache-http-mime"
                 MODULE="httpmime-4.1.2.jar" MVN="mvn:org.talend.libraries/httpmime-4.1.2/6.0.0" 
                 UrlPath="platform:/plugin/org.talend.libraries.apache.http/lib/httpmime-4.1.2.jar"
                	REQUIRED="true" />
             <IMPORT NAME="apache-commons-logging"
-                MODULE="commons-logging-1.1.1.jar" MVN="mvn:org.talend.libraries/commons-logging-1.1.1/6.0.0" 
-                UrlPath="platform:/plugin/org.talend.libraries.apache.common/lib/commons-logging-1.1.1.jar"
+                MODULE="commons-logging-1.2.jar" MVN="mvn:commons-logging/commons-logging/1.2" 
+            	REQUIRED="true" BundleID="" />
+            <IMPORT NAME="apache-commons-codec"
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBoxConnection/tBoxConnection_java.xml]: here also need to add comment on jira for  why import  new jars.

File: main/plugins/org.talend.designer.components.localprovider/components/tBonitaDeploy/tBonitaDeploy_java.xml (hunk 1 above)
Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tBonitaDeploy/tBonitaDeploy_java.xml]: This is httpmime, httpclient does not rely on it. I don't think we need to upgrade it.

**Pull request merged**